feat(react-grid): publish TableSelection's rowComponent (#2676)
DevExpress/devextreme-reactive: packages/dx-react-grid/docs/reference/table-selection.md

showSelectionColumn? | boolean | true | Specifies whether to render the selection column that displays selection checkboxes.
cellComponent | ComponentType&lt;[TableSelection.CellProps](#tableselectioncellprops)&gt; | | A component that renders a selection cell (a cell containing a selection checkbox).
headerCellComponent | ComponentType&lt;[TableSelection.HeaderCellProps](#tableselectioncellprops)&gt; | | A component that renders a cell containing the Select All checkbox.

> rowComponent | ComponentType&lt;[TableSelection.RowProps](#tableselectionrowprops)&gt; | | A component that renders a selection row (this component is rendered instead [Table.Row](table.md#tablerowprops) only when enabled `highlightRow` or `selectByRowClick`).

Reviewer: instead *of*

> rowComponent | ComponentType&lt;[TableSelection.RowProps](#tableselectionrowprops)&gt; | | A component that renders a selection row (this component is rendered instead of [Table.Row](table.md#tablerowprops) only when enabled `highlightRow` or `selectByRowClick`).

Reviewer 2: ```suggestion
rowComponent | ComponentType&lt;[TableSelection.RowProps](#tableselectionrowprops)&gt; | | A component that renders a selected row. It is used instead of [Table.Row](table.md#tablerowprops) when `highlightRow` or `selectByRowClick` is enabled.
```

selected | boolean | Indicates whether a row is selected.
onToggle | () => void | An event that initiates row selecting or deselecting.

### TableSelection.RowProps

> Describes properties passed to a component that renders a cell containing a selection checkbox.

Reviewer 2: ```suggestion
Properties passed to the `rowComponent`.
```

Extends [Table.RowProps](table.md#tablerowprops)

Field | Type | Description
------|------|------------

> selectByRowClick | boolean | Indicates if row can be selected by click.

Reviewer 2: ```suggestion
selectByRowClick | boolean | Indicates if the row can be selected by click.
```

> highlighted | boolean | Indicates whether a row is highlighted.

Reviewer 3: Maybe we should provide `selected` prop in addition to 'highlighted'? Because in case we do not highlight rows on select event `highlighted` will always be `false`.

> highlighted | boolean | Indicates whether a row is highlighted (enabled `highlightRow` and row is selected).

Reviewer 2: ```suggestion
highlighted | boolean | Indicates whether the row is highlighted. `true` when `highlightRow` is enabled and the row is selected.
```

> onToggle | () => void | An event that initiates row selecting or deselecting.

Reviewer 2: ```suggestion
onToggle | () => void | A function that is executed when users select the row or cancel the selection.
```
